model/renderNativeWindow.h

```diff
@@ -24,12 +24,44 @@
 
 namespace minsky
 {
-  class RenderNativeWindow: public ecolab::CairoSurface
+  struct WindowInformation
+  {
+    unsigned long parentWindowId;
+    unsigned long childWindowId;
+    int childWidth;
+    int childHeight;
+    int offsetLeft;
+    int offsetTop;
+    Display*	display;
```

Comment:
Because of the problems pointers create, it is worthwhile noting that this is a weak pointer, ie ownership of the display object remains with the X11 system.